### PR TITLE
fix: reCAPTCHA v2 + v3 for RAS-ACC flows

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes for reCAPTCHA implementation in RAS-ACC modal checkout.

### How to test the changes in this Pull Request:

See https://github.com/Automattic/newspack-plugin/pull/3283

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/modal-checkout/index.js [hunk 1]
```diff
@@ -391,6 +398,15 @@ domReady(
 						} );
 						$form.on( 'submit', handleFormSubmit );
 					} else {
+						if ( $form.find( '[name="is_validation_only"]' ) ) {
+							$form.find( '[name="is_validation_only"]' ).remove();
+						}
```
Comment (chickenn00dle) [inline src/modal-checkout/index.js]: Nonblocking nitpick: Maybe we can just create a new variable for the first call to `find` here so we don't need to run this twice.

```suggestion
                        const validationOnlyField = $form.find( '[name="is_validation_only"]' );
                        if ( validationOnlyField.length ) {
							$validationOnlyField.remove();
						}

```

**Review by chickenn00dle — APPROVED** [review]
Approving pending changes in the main plugin PR for this.

File: src/modal-checkout/index.js (hunk 1 above)
Comment (dkoo) [reply src/modal-checkout/index.js]: Great idea!

File: includes/class-modal-checkout.php [hunk 2]
```diff
@@ -528,18 +528,24 @@ class="<?php echo esc_attr( "$class_prefix {$class_prefix}__modal-container news
...
 			return;
 		}
 
 		if ( ! self::is_modal_checkout() ) {
 			return;
 		}
 
+		$dependencies = [ 'jquery' ];
+		// Add support reCAPTCHA dependencies, if connected.
+		if ( class_exists( 'Newspack\Recaptcha' ) ) {
```
Comment (chickenn00dle) [inline includes/class-modal-checkout.php]: We need to add a check for `\Newspack\Recaptcha::can_use_captcha()` here so we aren't loading the dependency when recaptcha is disabled.

**Review by chickenn00dle — DISMISSED** [review]

File: includes/class-modal-checkout.php (hunk 2 above)
Comment (chickenn00dle) [reply includes/class-modal-checkout.php]:
```suggestion
		if ( class_exists( 'Newspack\Recaptcha' ) && \Newspack\Recaptcha::can_use_captcha() ) {
```

Comment (dkoo) [reply includes/class-modal-checkout.php]: Aha!

**Review by chickenn00dle — APPROVED** [review]

**Pull request merged**